daos-stack/daos
# DAOS-9385 test: Add agent failure test (#8368)

### Additional Change
This is the cherry-pick PR, but deployment dir doesn't exist in
release/2.0 branch at this point, so added the entry in the list
in SConscript.

In addition, 100% size pool doesn't work in 2.0, so reduced the
size to 90%.
### Test Summary
Verify agent failure is properly handled.

1. Create a pool and a container.
2. Run IOR.
3. Stop daos_agent process while IOR is running.
4. Check the error on the client side. When daos_agent is killed
in the middle of IOR, the file write completes successfully, but
the pool disconnect at the end fails, so we get the error message
that includes -1005. This step is more like a verification of the
test itself rather than the product.
5. Verify daos_server doesn’t have the pool handle opened. Call:
dmg telemetry metrics query --metrics=engine_pool_pool_handles
that shows the number of open pool handle. Verify that it's 0 after
the agent is stopped and IOR is finished. This step verifies the
isolation of the failure. In this case, daos_agent was stopped,
but daos_server can detect it and close the pool handle.
6. Verify journalctl shows the log that the agent is stopped. Call:
journalctl --system -t daos_agent --since <before> --until <after>
This step verifies that DAOS, or daos_agent process in this case,
prints useful logs for the user to troubleshoot the issue, which
in this case the application can’t be used.
7. Restart daos_agent.
8. Run IOR again. It should succeed this time without any error.
This step verifies that DAOS can recover from the fault with
minimal human intervention.

Skip-unit-tests: true
Test-tag: agent_failure
Signed-off-by: Makito Kano <makito.kano@intel.com>
